### DISCUSSION TITLE
Searching for Sponsor(s) for LDAP & SSO implementation

### DISCUSSION THREAD

**Discussion by Kovah** [Kovah/LinkAce#671] [General]
Dear community,

I decided that I will not add support for LDAP or any SSO protocols to LinkAce yet. LinkAce was never intended to be used in companies or large communities where such authentication methods are helpful. 

**If companies want enterprise-level authentication, they will have to pay for it.** It's that easy. I am not going to work on such a large feature that I am not using at all, all for free, in my leisure time.

### My offer to anyone interested

Become a sponsor with a significant contribution, and LDAP and/or SSO will happen. The sponsor(s) will be mentioned with a link on the LinkAce website and in the README beside the corresponding feature, like _LDAP (sponsored by Acme Inc.)_. The feature will then become available for free to everyone.
I do not have a specific amount in mind, but it should correspond with the amount of work that needs to be done and the usual rates for software developers.

If you are interested, please contact me via `contact@linkace.org` or leave a comment here.

### What about LDAP / SSO as a paid add-on?

I thought about that and decided not to go in that direction. The amount of work needed to make this work is even worse than just getting paid to implement it and offer it for free to everyone. It doesn't help anyone.

---

PS: If you like LinkAce and use it daily, please consider sponsoring me and my work. 

- [Github sponsors](https://github.com/sponsors/Kovah)
- [Open Collective](https://opencollective.com/linkace)
- [Patreon](https://patreon.com/Kovah)

---

Related issues and discussions: 
- #174 
- #194
- #331
- #555

**Comment by electrofloat** [comment]
Although you need to understand that today with the help of Authelia, Authentik, Keycloak, etc. more and more home users are using ldap and sso. The time when these were enterprise only, is long gone.

**Reply by dkowis** [reply]
Agreed with this. [lldap](https://github.com/lldap/lldap) and Keycloak make this super easy. If I want to stay off the corporate clouds (google, amazon, etc.) and run my own services at home, it quickly becomes untenable to maintain user and authentication for every service individually. Especially if I run it for even just 2 users. 

I see Oauth2 as a way of encouraging running multiple services at home, because it cooperates with the rest of the authentication ecosystem. It makes it more likely that users will run multiple things at home and helps with making the web more decentralized again.

**Reply by Kovah** [reply]
I understand the reasoning and the point that LDAP and SSO are also interesting for home environments. However, I am not using either of both authentication systems, so I am not going to spend time working on it.

LinkAce is open source and I don't charge for it. So feel free to contribute the code for the feature if you are not able to sponsor the project to get it done.

**Reply by dkowis** [reply]
I understand that too! I'm glad we're on the same page :) Thanks for all your hard work!
